src/lib/message/L1Transaction.ts

```diff
@@ -58,6 +67,46 @@ export type L1EthDepositTransaction =
 export type L1ContractCallTransaction =
   L1ContractTransaction<L1ContractCallTransactionReceipt>
 
+class ClassicL1ToL2MessageReader implements ClassicL1ToL2Message {
+  public readonly messageNumber: BigNumber
+  public readonly retryableCreationId: string
+  public readonly autoRedeemId: string
+  public readonly l2TxHash: string
+  public readonly l2Provider: Provider
+
+  constructor(props: {
```

Comment:
Is the missing call to `super` intentional? Do we want to extends from `ClassicL1toL2Message`?

Reply:
This class implements ClassicL1toL2Message, doesn't extend it